csharp/test/Apache.Arrow.Tests/TestData.cs

```diff
@@ -48,7 +50,7 @@ public static RecordBatch CreateSampleRecordBatch(int length, int columnSetCount
                 builder.Field(CreateField(Date32Type.Default, i));
                 builder.Field(CreateField(Date64Type.Default, i));
                 builder.Field(CreateField(Time32Type.Default, i));
-                builder.Field(CreateField(Time64Type.Default, i));
```

Comment:
The time unit on Time64Type.Default seems like an odd choice in that it produces a type which can't be expressed via the C API.

Reply:
I'd like to understand this better. Why can't `TimeUnit.Millisecond` work with the C API?

Reply:
Time64 with milliseconds is not explicitly enumerated at https://arrow.apache.org/docs/format/CDataInterface.html and it would need a new format specifier to work. This is because "tt" is used for both Time32 and Time64 and the time unit is used to determine which one it will be. (This is perhaps an error in the specification, but the spec is what it is.)

Reply:
it looks like Tim32 can be seconds or milliseconds where as Time64 should be microseconds or nanoseconds.